crates/bevy_ecs/src/system/commands.rs

```diff
@@ -413,6 +423,19 @@ impl<T: Component> Command for RemoveResource<T> {
     }
 }
 
+pub struct RunSystem {
+    system: Mutex<RefCell<Box<dyn System<In = (), Out = ()>>>>,
+}
```

Comment:
So a few things with this.
1) I don't think we necessarily want to have this as `pub` since it's more of an implementation detail.

2) as a small optimization, we could optionally declare the struct as such:
```rust
struct RunSystem {
    system: UnsafeCell<Box<dyn System<In = (), Out = ()>>>,
}

unsafe impl Sync for RunSystem {}
```
IIUC, we can safely use `UnsafeCell` and `unsafe impl Sync` since we know no data races are possible as `Commands` have exclusive mutable access to the `World`. 

_Note_: I'm not necessarily advocating for the `unsafe` path, however, it is worth considering since then we don't have to worry about the overhead that `Mutex` and `RefCell` introduce.

Reply:
This is assuming you even need the `UnsafeCell` in the first place...?

Reply:
> I don't think we necessarily want to have this as pub since it's more of an implementation detail.

All commands are pub, so I would prefer to continue that way

> as a small optimization, we could optionally declare the struct as such

I would prefer to avoid unsafe unless it's a clear Ian in perf? But I don't think perf are a big issue for one off systems

Reply:
Cool, the `pub` thing makes sense!

Though now that we removed the `RefCell`, I'd still prefer the `unsafe impl Sync for RunSystem` and remove the `Mutex`, however I don't feel super strongly about this.